src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,172 @@
+        }
+        productInsert(shoppingCart);
+        return shoppingCart;
+    }
+
+    @Override
+    public boolean deleteById(Long cartId) {
+        String query = "UPDATE shopping_carts SET deleted = true WHERE cart_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, cartId);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("It is impossible to delete the buyer's card"
+                    + cartId, e);
+        }
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        List<ShoppingCart> shoppingCarts = new ArrayList<>();
+        String query = "SELECT * FROM shopping_carts WHERE deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                shoppingCarts.add(getShoppingCartFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all shopping carts", e);
+        }
+        for (ShoppingCart shoppingCart : shoppingCarts) {
+            shoppingCart.setProducts(getProductFromShoppingCart(shoppingCart.getId()));
+        }
+        return shoppingCarts;
+    }
+
+    private ShoppingCart getShoppingCartFromResultSet(ResultSet resultSet) throws SQLException {
+        Long cartId = resultSet.getLong("cart_id");
+        Long userId = resultSet.getLong("user_id");
+        return new ShoppingCart(cartId, userId);
+    }
+
+    private List<Product> getProductFromShoppingCart(Long shoppingCartId) {
+        String query = "SELECT * FROM products p "
+                + "INNER JOIN shopping_carts_products cp ON p.product_id = "
+                + "cp.product_id WHERE cp.cart_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, shoppingCartId);
+            ResultSet resultSet = statement.executeQuery();
+            List<Product> products = new ArrayList<>();
+            while (resultSet.next()) {
+                Long productId = resultSet.getLong("product_id");
+                String name = resultSet.getString("name");
+                BigDecimal price = resultSet.getBigDecimal("price");
+                products.add(new Product(productId, name, price));
+            }
+            return products;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get shopping cart with ID =" + shoppingCartId,
+                    e);
+        }
+    }
+
+    private ShoppingCart productInsert(ShoppingCart shoppingCart) {
```

Comment:
what about this method naming ? 
```suggestion
    private ShoppingCart insertProducts(ShoppingCart shoppingCart) {
```

Reply:
Sure. Sorry.